interview-questions.md

```diff
@@ -8,36 +8,108 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: Three rails generator commands include:
+rails g model - Creates a new model, which will accept key:value pairs of a name and a data type. This is essentially initializing the schema for an empty table, which we will then populate with our data.
+
+rails g migration - Generates a new instance of a migration, which we can use to create, edit, and destroy parts of our model.
+
+rails g resource - Generates a new resource, which is similar to a model.
 
-Researched answer:
+Researched answer: In addition to my previous response, the primary difference between a model and a resource is what folders they create when you initialize them. A model will create a migration, a model in app/models, and some tests. A resource will create a migration, a model in app/models, a controller, routes, and tests. So, a model could be useful for when you want more flexibility and customization with your app, but a resource will give you more resources on initialization so that you can hit the ground running.
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-action: "GET" location: /students
+action: "GET" location: /students -> index
+
+```ruby
+def index
+  @students = Student.all
+end
+```
+
+action: "POST" location: /students -> create
+
+```ruby
+def create
+  @student = Student.create(strong_params)
+end
+```
 
-action: "POST" location: /students
+action: "GET" location: /students/new -> new
 
-action: "GET" location: /students/new
+```ruby
+def new
+  @student = Student.new
+end
+```
 
-action: "GET" location: /students/2
+action: "GET" location: /students/2 -> show
 
-action: "GET" location: /students/2/edit
+```ruby
+def show
+  @student = Student.find(params[:id])
+end
+```
 
-action: "PATCH" location: /students/2
+action: "GET" location: /students/2/edit -> edit
 
-action: "DELETE" location: /students/2
+```ruby
+def edit
+  @student = Student.find(params[:id])
+end
+```
+
+action: "PATCH" location: /students/2 -> update
+
+```ruby
+def update
+  @student = Student.find(params[:id])
+  @student.update(strong_params)
+end
+```
+
+action: "DELETE" location: /students/2 -> delete
+
+```ruby
+def delete
+  @student = Student.find(params[:id]).destroy
+end
+```
 
 5. As a developer, you are making an application to manage your to do list. Create 10 user stories that will help you get your application started. Read more about [user stories](https://www.atlassian.com/agile/project-management/user-stories).
+
+1. As a user, I can see a webpage with a header that says 'To Do List', and a button that says 'Add Task'.
```

Comment:
Awesome work in this entire section.  Really detailed and thorough user stories 👏